test/jdk/javax/swing/JComboBox/ComboPopupBug.java

```diff
@@ -21,66 +21,74 @@
  * questions.
  */
 
+import java.awt.Dimension;
+import java.awt.Point;
+import java.awt.Robot;
+import java.awt.event.InputEvent;
 import javax.swing.JButton;
 import javax.swing.JComboBox;
 import javax.swing.JFrame;
+import javax.swing.SwingUtilities;
 
 /*
  * @test
  * @bug 8322754
+ * @key headful
  * @summary Verifies clicking JComboBox during frame closure causes Exception
- * @library /java/awt/regtesthelpers
- * @build PassFailJFrame
- * @run main/manual ComboPopupBug
+ * @library /javax/swing/regtesthelpers
+ * @run main ComboPopupBug
  */
 
 public class ComboPopupBug {
-    private static final String instructionsText = """
-            This test is used to verify that clicking on JComboBox
-            when frame containing it is about to close should not
-            cause IllegalStateException.
-
-            A JComboBox is shown with Close button at the bottom.
-            Click on Close and then click on JComboBox arrow button
-            to try to show combobox popup.
-            If IllegalStateException is thrown, test will automatically Fail
-            otherwise click Pass.""";
+    private static JFrame frame;
+    private static JButton closeButton;
+    private static JComboBox<String> comboBox;
+    private static Point comboBoxLocation;
+    private static Dimension comboBoxSize;
+    private static Robot robot;
+    private static final int PADDING = 10;
 
     public static void main(String[] args) throws Exception {
-        PassFailJFrame.builder()
-                .title("ComboPopup Instructions")
-                .instructions(instructionsText)
-                .testTimeOut(5)
-                .rows(10)
-                .columns(35)
-                .testUI(ComboPopupBug::createUI)
-                .build()
-                .awaitAndCheck();
-    }
+        robot = new Robot();
+        SwingUtilities.invokeAndWait(() -> {
+            frame = new JFrame("ComboPopup");
+
+            comboBox = new JComboBox<>();
+            comboBox.setEditable(true);
+            comboBox.addItem("test");
+            comboBox.addItem("test2");
+            comboBox.addItem("test3");
 
-    private static JFrame createUI() {
-        JFrame frame = new JFrame("ComboPopup");
```

Comment:
To reduce the number of changes compared to the previous version, I suggest keeping `createUI` method, just change its return type to `void` and pass the reference `ComboPopupBug::createUI` to `invokeAndWait`.

This way would be cleaner, I think.